### PR TITLE
⚗️✨ [RUMF-1379] heatmaps: enable descendant combined selectors

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

After experimenting with https://github.com/DataDog/browser-sdk/pull/1741 , we chose to keep only one of the two experiments.

## Changes

 This PR promote the experimental selector as the default one, and does a bit of cleanup and redesign.

## Testing

- [x] Local
- [x] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by bcaudan — APPROVED** [review]

**Review by liywjl — APPROVED** [review]

File: packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts [hunk 1]
```diff
@@ -22,88 +22,67 @@ export const STABLE_ATTRIBUTES = [
...
-      useCombinedSelectors ? targetElementSelector : undefined
-    )
-    targetElementSelector = combineSelector(
-      uniqueSelectorAmongChildren || getPositionSelector(element) || getTagNameSelector(element),
+      actionNameAttribute,
       targetElementSelector
     )
+    targetElementSelector =
+      uniqueSelectorAmongChildren ||
+      combineSelector(getPositionSelector(element) || getTagNameSelector(element), targetElementSelector)
```
Comment (amortemousque) [inline packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts:66]: **❓ question:** ‏Would it work like so:
```suggestion
      combineSelector(getPositionSelector(element), targetElementSelector)
```
It passes tests

Comment (BenoitZugmeyer) [reply packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts:66]: Hum that's right, it doesn't seem necessary. Let me improve the situation

**Comment by amortemousque** [comment]
**👏 praise:** ‏Very good job on this whole topic!

File: packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts (hunk 1 above)
Comment (BenoitZugmeyer) [reply packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts]: Done!

**Review by amortemousque — APPROVED** [review]

**Pull request merged**